### PR TITLE
Clean up InTotoVerify and rsa public key loading

### PR DESCRIPTION
**Pull request by lukpueh**
This PR provides a few small code refactors for clean up:
- Summarize common functionality used in keylib VerifySignature and LoadPublicKey #7 
- Move type juggeling loops from InTotoVerify to helper methods #6 
- Implement go-thonic field inheritance for common fields of steps/inspections

### CODE REVIEW

File: in_toto/keylib.go [hunk 1]
```diff
@@ -14,6 +14,29 @@ import (
   "reflect"
 )
 
+func ParseRSAPublicKeyFromPEM(pemBytes []byte) (*rsa.PublicKey, error) {
+  // TODO: There could be more key data in _, which we silently ignore here.
+  // Should we handle it / fail / say something about it?
```
Comment (SantiagoTorres) [inline in_toto/keylib.go]: If we were to merge this, could we ticketize this issue?

File: in_toto/model.go [hunk 2]
```diff
@@ -68,6 +68,27 @@ type Layout struct {
   Readme string `json:"readme"`
 }
 
+// Go does not allow to pass `[]T` (slice with certain type) to a function
+// the accepts `[]interface{}` (slice with generic type)
```
Comment (SantiagoTorres) [inline in_toto/model.go]: s/the/that/

File: in_toto/model.go [hunk 3]
```diff
@@ -68,6 +68,27 @@ type Layout struct {
...
+  }
+  return stepsI
+}
+func (l *Layout) InspectAsInterfaceSlice() []interface{} {
+  inspectionsI := make([]interface{}, len(l.Inspect))
+  for i, v := range l.Inspect {
+    inspectionsI[i] = v
+  }
+  return inspectionsI
+}
```
Comment (SantiagoTorres) [inline in_toto/model.go]: Aside: is this just another instance of lolnogenerics?

File: in_toto/keylib.go (hunk 1 above)
Comment (lukpueh) [reply in_toto/keylib.go]: Yes. Done. #14.

File: in_toto/model.go (hunk 3 above)
Comment (lukpueh) [reply in_toto/model.go]: :)

File: in_toto/model.go (hunk 2 above)
Comment (lukpueh) [reply in_toto/model.go]: a685b474ff64e059aabbc1feb125c899305db28c

**Comment by lukpueh** [comment]
@SantiagoTorres, thanks for your review! I addressed you comments, except for the one about lolnogenerics, I fear I can't fix that one. :P  I also had to quickly fix the test layout expiration (#15) and rebase.

**Comment by SantiagoTorres** [comment]
Thanks @lukpueh !

**Pull request merged**